src/NewTwitchApi/Resources/EventSubApi.php

```diff
@@ -0,0 +1,286 @@
+        );
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#channelcheer
+     */
+    public function subscribeToChannelCheer(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribe(
+            'channel.cheer',
+            '1',
+            ['broadcaster_user_id' => $twitchId],
+            $callback,
+            $bearer,
+            $secret
+        );
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#channelraid
+     */
+    public function subscribeToChannelRaid(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribe(
+            'channel.raid',
+            '1',
+            ['broadcaster_user_id' => $twitchId],
+            $callback,
+            $bearer,
+            $secret
+        );
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#channelban
+     */
+    public function subscribeToChannelBan(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribe(
+            'channel.ban',
+            '1',
+            ['broadcaster_user_id' => $twitchId],
+            $callback,
+            $bearer,
+            $secret
+        );
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#channelunban
+     */
+    public function subscribeToChannelUnban(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribe(
+            'channel.unban',
+            '1',
+            ['to_broadcaster_user_id' => $twitchId],
+            $callback,
+            $bearer,
+            $secret
+        );
+    }
+
+    private function subscribeToChannelModerator(string $twitchId, string $event, string $callback, string $bearer, string $secret): void
+    {
+        if (!in_array($event, ['add', 'remove'])) {
+            throw new \InvalidArgumentException('Invalid value for channel.moderator event type. Accepted values: add,remove.');
+        }
+
+        $this->subscribe(
+            sprintf('channel.hype_train.%s', $event),
+            '1',
+            ['broadcaster_user_id' => $twitchId],
+            $callback,
+            $bearer,
+            $secret
+        );
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#channelmoderatoradd
+     */
+    public function subscribeToChannelModeratorAdd(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribeToChannelModerator($twitchId, 'add', $callback, $bearer, $secret);
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#channelmoderatorremove
+     */
+    public function subscribeToChannelModeratorRemove(string $twitchId, string $callback, string $bearer, string $secret): void
```

Comment:
- Bearer should always be the first parameter
- Since Secrets are required for all Subscriptions, put this directly after the Bearer
- Since Callbacks are required for all Subscriptions, put this after the Secret